### PR TITLE
XRT-1145 Update to represent infinite floating point values in JSON

### PR DESCRIPTION
**Pull request by SteveOss**

### CODE REVIEW

File: src/c/data.c [hunk 1]
```diff
@@ -1354,8 +1355,18 @@ static void iot_data_dump_raw (iot_string_holder_t * holder, const iot_data_t *
...
+    case IOT_DATA_FLOAT32:
+    {
+      float f = iot_data_f32 (data);
+      (fpclassify (f) == FP_INFINITE) ? snprintf (buff, IOT_VAL_BUFF_SIZE, "1e400") : snprintf (buff, IOT_VAL_BUFF_SIZE, "%.8e", f);
+      break;
+    }
+    case IOT_DATA_FLOAT64:
+    {
+      double d = iot_data_f64 (data);
+      (fpclassify (d) == FP_INFINITE) ? snprintf (buff, IOT_VAL_BUFF_SIZE, "1e800") : snprintf (buff, IOT_VAL_BUFF_SIZE, "%.16e", d);
```
Comment (FraserG01) [inline src/c/data.c:1367]: How can we go past 1e500 (infinity) to 1e800?

Comment (SteveOss) [reply src/c/data.c:1367]: Yes, it's like infinity only bigger :) Essentially there is a eXXXX upper limit for the representation of float values (different
for float and double) after which anything is just regarded as 'infinite' so any number larger than the limit works, I picked
different representations for float/double (based on byte size of target data), so if you see one generated in json you
can deduce the original data type.

Comment (FraserG01) [reply src/c/data.c:1367]: ah right okay makes sense

**Review by FraserG01 — APPROVED** [review]
LGTM

**Pull request merged**